src/FileSystemActions.ts

```diff
@@ -140,15 +140,17 @@ export default class FileSystemActions {
                 // revert before delete in case it's optn for add/edit.  At
                 // come point maybe dialog to warn user but this does
                 // match logic in PerforceCommands
-                PerforceCommands.p4revert(uri);
-                PerforceCommands.p4delete(uri);
+                await PerforceCommands.p4revert(uri);
+                await PerforceCommands.p4delete(uri);
             }
         }
+        PerforceSCMProvider.RefreshAll();
```

Comment:
`HandleCommonServiceResponse` also calls `Display.updateEditor();` which updates the status icon in the status bar

Reply:
added to p4revert/p4delete calls